streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -211,28 +207,44 @@ public synchronized void start() {
      * and then wait for them to join.
      *
      * @throws IllegalStateException if process has not started yet
+     * @param timeout   how long to wait for {@link StreamThread}s to shutdown
+     * @param timeUnit  unit of time used for timeout
      */
-    public synchronized void close() {
+    public synchronized void close(final long timeout, final TimeUnit timeUnit) {
```

Comment:
@enothereska if you are trying to shutdown an application in a timely manner. First you may try and shutdown cleanly, but you'll have a limit on how long you want to wait before you forceably kill the application. Maybe that timeout is 1 minute, maybe it is 10 seconds, point being that we can't know what the user needs.  It should never be the case that you just call close it and potentially blocks forever.